File: app/src/main/java/com/temperoni/recipes/component/RecipesComponent.java
package com.temperoni.recipes.component;

import com.temperoni.recipes.ui.activities.RecipeDetailActivity;
import com.temperoni.recipes.ui.activities.RecipesListActivity;

import dagger.Subcomponent;

/**
 * @author Leandro Temperoni
 */
@Subcomponent
public interface RecipesComponent {

    void inject(RecipesListActivity recipesListActivity);

    void inject(RecipeDetailActivity recipeDetailActivity);
}

File: app/src/main/java/com/temperoni/recipes/component/RecipesComponent.kt
package com.temperoni.recipes.component

import com.temperoni.recipes.ui.activities.RecipeDetailActivity
import com.temperoni.recipes.ui.activities.RecipesListActivity

import dagger.Subcomponent

/**
 * @author Leandro Temperoni
 */
@Subcomponent
interface RecipesComponent {

    fun inject(recipesListActivity: RecipesListActivity)

    fun inject(recipeDetailActivity: RecipeDetailActivity)
}
